src/test/java/com/iexec/blockchain/tool/ChainConfigTest.java

```diff
@@ -0,0 +1,129 @@
+package com.iexec.blockchain.tool;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import javax.validation.ConstraintViolationException;
+import javax.validation.Validation;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class ChainConfigTest {
+    private ChainConfig chainConfig;
+
+    @BeforeEach
+    void setUp() {
+        this.chainConfig = new ChainConfig(
+                Validation.buildDefaultValidatorFactory().getValidator()
+        );
+    }
+
+    static Stream<Arguments> validData() {
+        return Stream.of(
+                Arguments.of(100, "http://localhost:8545", "0xBF6B2B07e47326B7c8bfCb4A5460bef9f0Fd2002", 1),
+                Arguments.of(42, "https://localhost:8545", "0x0000000000000000000000000000000000000000", 10),
```

Comment:
This should not be valid. The contract address should not be `0x0000000000000000000000000000000000000000`.

Reply:
This means that `@ValidEthereumAddress` should probably be `@ValidNonZeroEthereumAddress`.

Reply:
I didn't know but seems great. I'll add this restriction, thanks! :)

Reply:
Might help: https://github.com/iExecBlockchainComputing/iexec-common/blob/06eedcc20bb85e0285ae07c1b7b04ddc9bfe138d/src/main/java/com/iexec/common/utils/BytesUtils.java#L85